go-install: add page (#7111)
tldr-pages/tldr: pages/common/go-install.md

# go install

> > Compile and install packages named by the import paths.

Author: `go install` places binaries in `$GOPATH/bin`, or in `$GOBIN` if set. May be worth specifying it, as is probably not clear for non-go developers.

The change you are proposing is a bit confusing; the import path is the argument to the `go install` command, for example: `go install filippo.io/mkcert@v1.4.2`.
Does this make it clearer?

> > Compile and install packages named by the import paths.

Reviewer: We should probably add a second description line to clear this up a bit. As a non-go developer, it sounds very confusing.

> > Compile and install packages named by the import paths.

Reviewer: We should probably add a second description line to clear this up a bit. As a non-go developer, it sounds very confusing.

> > Compile and install packages named by the import paths.

Author: Let me try explaining how this works and some context, than we can summarize what is more meaningful. As a Go developer these are the basics, so I need some help from someone unfamiliar to select the relevant bits.

Go has the concept of _workspace_, a folder within which all Go code was expected to be (this changed in Go `1.11`). `GOPATH` is the environment variable that specify the root folder of a Go workspace.
Within a Go workspace there are 2 key folders, `src` and `bin`. `src` contains all Go sources, `bin` contains all executable commands.

Since Go `1.11` a new way to manage Go code was introduced, _modules_. A _module_ is a folder containing Go code and a `go.mod` file that exists outside of a workspace (so outside of `GOPATH`).
Modules from within a `go.mod` are installed in the `pkg` folder in your workspace (so `$GOPATH/pkg`); that is **source** files are within that folder.

`go install` download a module, compiles it and move the resulting executable in the appropriate folder. Executables are installed in the directory named by the `GOBIN` environment variable, which defaults to `$GOPATH/bin` or `$HOME/go/bin` if the `GOPATH` environment variable is not set.

Hope it's clearer now :)

> > Compile and install packages named by the import paths.

Reviewer: Thanks for the explanation. I'm not sure either how to break this down into one sentence. I think we can leave it like is and if there's still room for improvement, someone else will do it sometime.

> More information: <https://pkg.go.dev/cmd/go#hdr-Compile_and_install_packages_and_dependencies>.

- Compile and install the current package:

`go install`

> - Compile and install a local package:

Reviewer: ```suggestion
- Compile and install a specific local package:
```

> `go install {{./cmd/...}}`

Reviewer: Is this correct?
```suggestion
`go install {{path/to/package}}`
```

- Install the latest version of a program, ignoring `go.mod` in the current directory:

`go install {{golang.org/x/tools/gopls}}@{{latest}}`

> - Install a program at the version selected by the module in the current directory:

Reviewer: `go.mod` from the previous example is the module right? I think we sholud keep that uniform to avoid confusion
```suggestion
- Install a program at the version selected by `go.mod` in the current directory:
```
